client/tests/rest_authentication_tests/common.py

```diff
@@ -0,0 +1,99 @@
+"""This class contains all the common methods required for basic and token based authentication."""
+
+import json
+
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.test import TestCase
+from rest_framework.authtoken.models import Token
+from rest_framework.test import APIClient
+
+from client.forms import ClientForm, MaleMeasurementsForm
+from client.models import Client, MaleMeasurements
+from client.utils import get_url_for_test_against_endpoint
+
+
+class ClientTestCase(TestCase):
+    """Common test case class having methods required in both basic and token auth tests."""
+
+    def setUp(self):
+        """Create a dummy user for the purpose of testing."""
+        self.username = 'test_user'
+        self.password = '12345678'
+        self.user, p = User.objects.get_or_create(username=self.username)
+        self.user.set_password(self.password)
+        self.user.save()
+        self.client_email = 'jon@nightwatch.com'
+        self.client_phone = '+9290078601'
+        self.client = Client.objects.create(**{
+            'name': 'Jon Snow',
+            'age': 18,
+            'gender': 'M',
+            'address': 'night watch wall',
+            'phone_number': self.client_phone,
+            'email': self.client_email
+        })
+        self.client.save()
+        self.api_client = APIClient()
```

Comment:
As force authentication client is used only in basic auth. you can move it to a function 'get_authenticated_client' in that child class.